### DISCUSSION TITLE
How to let mypy allow extra fields?

### DISCUSSION THREAD

**Discussion by ziyuang** [pydantic/pydantic#7352] [Question] [answered]
To reproduce with pydantic v1:

```python
from pydantic import BaseModel, Extra


class Model(BaseModel, extra=Extra.allow):
    ...


model = Model(a=1, b=2)
print(model.a)
```

mypy will complain `error: "Model" has no attribute "a"  [attr-defined]`. I have enabled the pydantic plugin in mypy.ini:

```
[mypy]
plugins = pydantic.mypy

```

**Comment by sydney-runkle** [comment, marked as answer]
Hi, thanks for your question.

The fact that you're using the `pydantic.mypy` plugin is silencing these two errors:
```
error: Unexpected keyword argument "a" for "Model"  [call-arg]
error: Unexpected keyword argument "b" for "Model"  [call-arg]
```

However, if you also would like to ignore the error you're experiencing, you could use the following.

```py
print(model.a)  # type: ignore[attr-defined]
```
